### PR TITLE
Incursion and Mission Ledger support

### PR DESCRIPTION
**Pull request by dysath**
This PR along with PR #88 from eveseat/web adds the Mission and Incursion Ledger support to SEAT.

### CODE REVIEW

File: src/Repositories/Corporation/Ledger.php [hunk 1]
```diff
@@ -133,4 +174,72 @@ public function getCorporationLedgerPITotalsByMonth(int $corporation_id,
...
+                    'MONTH(date) as month, YEAR(date) as year, ' .
+                    'ROUND(SUM(amount)) as total, ownerName1, ownerID1'
+                ))
+            ->where('corporationID', $corporation_id)
+            ->where('ownerName1', 'CONCORD')
+            ->where(DB::raw('YEAR(date)'), ! is_null($year) ? $year : date('Y'))
+            ->where(DB::raw('MONTH(date)'), ! is_null($month) ? $month : date('m'))
+            ->where(function ($query) {
+
+                $query->where('refTypeID', 99);
```
Comment (warlof) [inline src/Repositories/Corporation/Ledger.php]: Send it as method parameter?

File: src/Repositories/Corporation/Ledger.php [hunk 2]
```diff
@@ -133,4 +174,72 @@ public function getCorporationLedgerPITotalsByMonth(int $corporation_id,
...
+                DB::raw(
+                    'MONTH(date) as month, YEAR(date) as year, ' .
+                    'ROUND(SUM(amount)) as total, ownerName1, ownerID1'
+                ))
+            ->where('corporationID', $corporation_id)
+            ->where(DB::raw('YEAR(date)'), ! is_null($year) ? $year : date('Y'))
+            ->where(DB::raw('MONTH(date)'), ! is_null($month) ? $month : date('m'))
+            ->where(function ($query) {
+
+                $query->where('refTypeID', 33)
```
Comment (warlof) [inline src/Repositories/Corporation/Ledger.php]: Send it as method parameter (an array of reftypeid?)

File: src/Repositories/Corporation/Ledger.php [hunk 3]
```diff
@@ -72,6 +72,47 @@ public function getCorporationLedgerPIDates(int $corporation_id): Collection
...
+     * @return \Illuminate\Support\Collection
+     */
+    public function getCorporationLedgerIncursionDates(int $corporation_id): Collection
+    {
+
+        return DB::table('corporation_wallet_journals')
+            ->select(DB::raw('DISTINCT MONTH(date) as month, YEAR(date) as year'))
+            ->where('corporationID', $corporation_id)
+            ->where('ownerName1', 'CONCORD')
+            ->where('refTypeID', 99)
```
Comment (warlof) [inline src/Repositories/Corporation/Ledger.php]: Add it as a method parameter?

File: src/Repositories/Corporation/Ledger.php [hunk 4]
```diff
@@ -72,6 +72,47 @@ public function getCorporationLedgerPIDates(int $corporation_id): Collection
...
+     */
+    public function getCorporationLedgerMissionDates(int $corporation_id): Collection
+    {
+
+        return DB::table('corporation_wallet_journals')
+            ->select(DB::raw('DISTINCT MONTH(date) as month, YEAR(date) as year'))
+            ->where('corporationID', $corporation_id)
+            ->where(function ($query) {
+
+                $query->where('refTypeID', 33)
```
Comment (warlof) [inline src/Repositories/Corporation/Ledger.php]: Send it as a method parameter?

**Comment by warlof** [comment]
By adding refTypeIds as a parameter of array type, you can merge all methods in a single one

**Comment by dysath** [comment]
Closing for replacement

**Pull request closed without merging**